kazoo/recipe/lock.py

```diff
@@ -167,14 +181,15 @@ def _inner_acquire(self, blocking, timeout):
                 # node was removed
                 raise ForceRetryError()
 
-            if self.acquired_lock(children, our_index):
+            predecessor = self.predecessor(children, our_index)
+            if not predecessor:
                 return True
 
             if not blocking:
                 return False
 
             # otherwise we are in the mix. watch predecessor and bide our time
-            predecessor = self.path + "/" + children[our_index - 1]
+            predecessor = self.path + "/" + predecessor
```

Comment:
While your at it how about u use the following:

https://github.com/python-zk/kazoo/blob/master/kazoo/protocol/paths.py#L19